Sources/MongoSwift/MongoCursor.swift

```diff
@@ -118,7 +118,13 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
     public func next() -> T? {
         do {
             let operation = NextOperation(cursor: self)
-            let out = try operation.execute()
+            // TODO SWIFT-374: we temporarily use a fake connection here. eventually MongoCursor will store its source
+            // connection and we can pass that in here instead (though the execute method still won't use it.)
+            // we want to pass in a specific connection and call execute() ourselves rather than using an operation
+            // executor as that would require checking out a separate, unused connection for every single next() call.
+            // the force unwrap is safe as this bitPattern is always valid.
+            // swiftlint:disable:next force_unwrapping
+            let out = try operation.execute(using: Connection(OpaquePointer(bitPattern: 1)!), session: self._session)
```

Comment:
Consider making a base `Connection` protocol so that we can pass in a `FakeConnection`/`DummyConnection` here, and possibly mock out connections passed in for operation execution.

Reply:
we will need to be able to read the underlying `mongoc_client_t` property out of real passed in `Connection`s, so I guess we would end up with something like this then? : 

```swift
/// A connection to the database.
internal protocol Connection {
    /// Pointer to underlying state.
    let handle: OpaquePointer
}

/// A connection to mongoc.
internal struct MongocConnection: Connection {
    /// Pointer to the underlying `mongoc_client_t`.
    internal let handle: OpaquePointer

    internal init(_ handle: OpaquePointer) {
        self.handle = handle
    }
}

/// A mock connection.
internal struct DummyConnection: Connection {
    internal let handle = OpaquePointer(bitPattern: 1)!
}
```

Reply:
something slightly weird about this is that our `ConnectionPool` is really then just a `MongocConnectionPool` 🤔 I guess maybe eventually `ConnectionPool` would become a protocol and then we could have a mock pool implementation?
since this line is going to disappear in a subsequent PR anyway i wonder if we should wait to introduce this stuff until we will actually need/use a `DummyConnection`

Reply:
If one of your next PRs is going to remove the dummy connection stuff anyways then I don't see a big need for including any of the added machinery around it. If as part of the design we'll continue to use dummy connections until say, we remove libmongoc, then I'm for including the `DummyConnection` thing.

Reply:
Let's defer this decision